docs: refine quick start (#487)
algorandfoundation/algokit-cli: docs/tutorials/intro.md

This will launch a guided menu system to create a specific project tailored to your needs. The templates are basic starter applications for various Algorand development scenarios. To read more about templates checkout AlgoKit detailed documentation.

For now we'll use the `python` template, which is a lightweight starting point for learning Algorand smart contract development using Python. You can initialise a project using this template by running:

```shell

> algokit init -t python --no-workspace -a preset_name "starter"

Reviewer: Looks like my previous update was never synced to the docs site.

> algokit init -t python --no-workspace -a preset_name "starter"

Reviewer: I'm wondering if we should just use the default and init with a workspace, which would shorten the command to `algokit init -t python -a preset_name "starter"`

Author: refreshed a screenshot and added a bit more verbose explanation on the structure that user will see with workspace mode by default

```

Next, you will be prompted for the name of your project. Finally, select the default value for the rest of the prompts (enter).

> Once finished, (if you have it installed) VS Code should automatically be opened with the initialised project and you will be prompted to install appropriate VS Code extensions. This starter app will contain one smart contract (built using the [Algorand Python](https://algorandfoundation.github.io/puya/) language) named `contract.py`, in the `smart_contracts/hello_world` folder, with one method (`hello`) that takes a `String` and returns a `String`. Notice that within the contract folder, there is a `deploy_config.py`, this defined the parameters for deployment of your smart contract to the select target chain (Algorand TestNet, Algorand MainNet, Algorand LocalNet or else).

Reviewer: ```suggestion
Once finished, (if you have it installed) VS Code should automatically be opened with the initialised project and you will be prompted to install appropriate VS Code extensions. This starter app will contain one smart contract (built using the [Algorand Python](https://algorandfoundation.github.io/puya/) language) named `contract.py`, in the `smart_contracts/hello_world` folder, with one method (`hello`) that takes a `String` and returns a `String`. Notice that within the contract folder, there is a `deploy_config.py` which defines the parameters for deployment of your smart contract to the target chain (Algorand TestNet, Algorand MainNet, Algorand LocalNet or else).
```

![AlgoKit Playground Contract](../imgs/algokitplayground.png)

## AlgoKit Project structure 🏗

> Since you have just instantiated an official AlgoKit template, each official smart contract template includes an interactive codespace walkthrough powered by [CodeTours](https://marketplace.visualstudio.com/items?itemName=vsls-contrib.codetour).

Reviewer: ```suggestion
Since you have just instantiated an official AlgoKit template, each official smart contract template includes an interactive codespace walkthrough powered by [CodeTour](https://marketplace.visualstudio.com/items?itemName=vsls-contrib.codetour).
```

To start the interactive walkthrough, install the extension and click on the green play button in the left bottom corner of the `Explorer` pane as demonstrated on the screenshot below (See `Getting Started with Your AlgoKit Project`).

![CodeTour Play Button](../imgs/algokitcodetour.png)

> Upon execution, the walkthrough will guide you through the key components of your fresh AlgoKit project instance, which is an important prerequisite to learn before running and deploying your first smart contract. Certainly, not because its hard to learn, but because it will save you a lot of time and efforts if you set the right foundation first.

Reviewer: ```suggestion
Upon execution, the walkthrough will guide you through the key components of your fresh AlgoKit project, which is an important prerequisite to learn before running and deploying your first smart contract. Certainly, not because its hard to learn, but because it will save you a lot of time and effort having the correct foundations.
```

## Run the Demo Application 🏃‍♂️

Once the `Algorand Python` based project is created, you can get started by building and deploying your first smart contract. There are a few possible ways to do so as demonstrated on the following video.

[![Watch the video](https://ipfs.algonode.xyz/ipfs/Qmc9mRaPoDyhUFmek4ETxVfKUKzUg9pf3Tss5xwkBGdQis)](https://ipfs.algonode.xyz/ipfs/Qma6gNqxsSFc9Jbh8kBTZyVLv5gqFj1xnrsjoeT6MAAwCw/)

> As shown above, click on clicking on `Run and Debug` and then `Build & Deploy contracts`. Alternatively, you can hit F5 (or whatever keyboard shortcut is set in VS Code for running the debugger). While you are viewing the `contract.py` file - it will:

Reviewer: ```suggestion
As shown above, click on `Run and Debug` and then `Build & Deploy contracts`. Alternatively, you can hit F5 (or whatever keyboard shortcut is set in VS Code for running the debugger). While you are viewing the `contract.py` file - it will:
```

1. Start LocalNet
2. Build the smart contract. Notice how a folder named `artifacts` is created with the TEAL and JSON files for the smart contract.
3. Deploy and call the `HelloWorld` smart contract (from `contract.py` file).

> After execution, the `App ID` of of the deployed contract and its Algorand address are displayed, followed by the message returned from the smart contract call (`Hello, world`).

Reviewer: ```suggestion
After execution, the `App ID` of the deployed contract and its Algorand address are displayed, followed by the message returned from the smart contract call (`Hello, world`).
```

At this point you have deployed a simple contract to a LocalNet Algorand network (running in your Docker environment) and called it successfully!

> Additionally, you can find the native TEAL smart contract code and the appropriate smart contract manifest JSON files have been output to the `artifacts` folder. Note that in this example, TEAL code is compiled using `PuyaPy` compiler, which is what `Algorand Python` relies on to represent your Python smart contract code in TEAL ([Transaction Execution Approval Language](https://developer.algorand.org/docs/get-details/dapps/avm/teal/)).

Reviewer: ```suggestion
Additionally, you can find the native TEAL smart contract code and the appropriate smart contract manifest JSON files have been output to the `artifacts` folder. Note that in this example, TEAL code is compiled using the `PuyaPy` compiler, which is what `Algorand Python` relies on to compile your Python smart contract code to TEAL ([Transaction Execution Approval Language](https://developer.algorand.org/docs/get-details/dapps/avm/teal/)).
```

Author: @neilcampbell remove excessive usage of word TEAL in favour of AVM bytecode

### Using AlgoKit CLI to build and deploy contracts 🛠️

> When using official AlgoKit templates via `algokit init`, in addition to integration with VSCode and JetBrains IDEs, you can manage your projects using the AlgoKit CLI at the convienence of your terminal.

Reviewer: ```suggestion
When using official AlgoKit templates via `algokit init`, you can manage your projects using the AlgoKit CLI at the convenience of your terminal.
```

> For example, to achieve the same results as in the demo earlier, build and deploy the `HelloWorld` contract from the `contract.py` file, you can run the following command from the terminal:

Reviewer: ```suggestion
For example, to achieve the same results as in the demo earlier, build and deploy the `HelloWorld` contract from the `contract.py` file by running the following command from the terminal:
```

```shell
algokit project run build
```

> Which will recompile your python contracts from `contract.py` into TEAL and related contract artifacts.

Reviewer: ```suggestion
This command will recompile your python contracts from `contract.py` and produce the necessary artefacts for deploying your contract.
```

```shell
algokit project deploy localnet
```

> This will then execute the deploy script to LocalNet instance and display the same output as in the earlier demo.

Reviewer: ```suggestion
This will then deploy to your LocalNet instance and display the same output as in the earlier demo.
```

## Next steps 🚶‍♂️

> We have only covered a tiny fraction of capabilities of the AlgoKit CLI and its related ecosystem of templates and utilities for efficient developer experience.

Reviewer: ```suggestion
We have only covered a tiny fraction of the capabilities of the AlgoKit CLI and its related ecosystem of templates and utilities for an efficient developer experience.
```

- To get the most out of `AlgoKit`, we recommend to get started with learning more about AlgoKit and what you can do with it by checkout our extensive [AlgoKit CLI documentation](../algokit.md).
- Explore the `README.md` files at the roots of projects created via `algokit init`. All official AlgoKit templates include detailed quick started guides, various presets and degrees of customizability, as well as an interactive reviews of the codebase.
- To learn more about `Algorand Python`, take a look at the [documentation](https://algorandfoundation.github.io/puya/).
- To learn more about commands demonstrated in this tutorial, refer to [`init`](../features/init.md) and [`project`](../features/project.md) to get a comprehensive understanding of their further capabilities.
- If you'd like to structure your `AlgoKit Project` as a monorepo, refer to `workspace` mode as described in [`init`](../features/init.md#workspaces-vs-standalone-projects).

> - If you'd like to **create your own** `AlgoKit` template, refer to the [dedicated tutorials](./algokit-template.md).

Reviewer: ```suggestion
- If you'd like to **create your own** `AlgoKit` template, refer to the [template tutorial](./algokit-template.md).
```
